test/unit/PositionManager.spec.ts

```diff
@@ -1,502 +1,508 @@
-//                 size: 50,
-//                 pips,
-//                 pipsHasLiquidity: [false, false, false, false, false, true, true],
-//                 reachPip: 254,
-//                 orders,
-//                 partialFilledAmounts: [...pips].fill(0),
-//                 isFilledAmounts: [true, true, true, true, true, false, false],
-//                 isBuy: false
-//             })
-//         });
-//     });
-// });
-//
+import {ethers} from 'hardhat';
+import {BigNumber} from 'ethers'
+import {expect} from 'chai';
+import {PositionManager} from "../../typeChain/PositionManager"
+import {BEP20Mintable} from "../../typeChain";
+
+describe('Position Manager', async function () {
+    let deployer: any;
+    let positionManager: PositionManager;
+    let bep20Mintable: BEP20Mintable
+    beforeEach(async () => {
+        [deployer] = await ethers.getSigners()
+
+        // Deploy mock busd contract
+        const bep20MintableFactory = await ethers.getContractFactory('BEP20Mintable')
+        bep20Mintable = (await bep20MintableFactory.deploy('BUSD Mock', 'BUSD')) as unknown as BEP20Mintable
+
+        const factory = await ethers.getContractFactory("PositionManager")
+        positionManager = (await factory.deploy()) as unknown as PositionManager
+        await positionManager.initialize(BigNumber.from(200), bep20Mintable.address, ethers.utils.formatBytes32String('BTC'), BigNumber.from(100), BigNumber.from(10000), BigNumber.from(10000), BigNumber.from(3000), BigNumber.from(1000), '0x5741306c21795FdCBb9b265Ea0255F499DFe515C'.toLowerCase(), deployer.address);
+    })
+
+    async function createLimitOrder(pip: number, size: number, isBuy: boolean) {
+        return positionManager.openLimitPosition(pip, size, isBuy);
+    }
+
+    async function createLimitOrderAndVerify(pip: number, size: number, isBuy: boolean) {
+        const {liquidity: liquidityBefore} = await positionManager.tickPosition(pip)
+        const pipPositionData = await positionManager.tickPosition(pip)
+        const orderId = pipPositionData.currentIndex.add(1)
+        await createLimitOrder(pip, size, isBuy);
+        console.log(`Create limit order ${isBuy ? 'BUY' : 'SELL'}`, orderId.toNumber(), `SIZE: ${size} at ${pip}`)
+        const hasLiquidity = await positionManager.hasLiquidity(pip)
+        expect(hasLiquidity).eq(true, `Pip #${pip}`)
+        const {liquidity} = await positionManager.tickPosition(pip)
+        expect(liquidity.sub(liquidityBefore).toNumber()).eq(size)
+        const orderDetail = await positionManager.getPendingOrderDetail(pip, orderId.toNumber());
+        expect(orderDetail.size.toNumber(), "size not match").eq(size)
+        return orderId.toNumber()
+    }
+
+    async function marketBuy(size: number, isBuy: boolean = true) {
+        return positionManager.openMarketPosition(size, isBuy)
+    }
+
+    const shouldBuyMarketAndVerify = async function (size: number, expectOut: number, isBuy: boolean = true) {
```

Comment:
change expect data from event "Swap" to event "MarketFilled". event "Swap" was removed in position manager